src/Types/ScriptLookups.purs

```diff
@@ -301,29 +289,23 @@ instance Semigroup ValueSpentBalances where
 type ConstraintProcessingState =
   { unbalancedTx :: UnbalancedTx
   -- The unbalanced transaction that we're building
-  , mintRedeemers :: Map MintingPolicyHash Redeemer
-  -- Redeemers for minting policies
   , valueSpentBalancesInputs :: ValueSpentBalances
   -- Balance of the values given and required for the transaction's inputs
   , valueSpentBalancesOutputs :: ValueSpentBalances
   -- Balance of the values produced and required for the transaction's outputs
   , datums :: Array Datum
-  -- *Set* of accumulating datums so we can use `setScriptDataHash`
-  , redeemers :: Map RedeemerTag (Array Redeemer)
-  -- Map of accumulating redeemers separated by `RedeemerTag`. Note that adding
-  -- to this discards the ordering of redeemers from constraints (should that
-  -- matter).
+  -- Ordered accumulation of datums so we can use to `setScriptDataHash`
+  , redeemers :: Array (RedeemerTag /\ T.Redeemer)
```

Comment:
I can't think of why we might need the `RedeemerTag`. Is there a specific future scenario you're thinking of? These are only being stored for `setScriptDataHash`, right? We always need to include all redeemers when calling that (AFAIK), so I don't think we'll need to distinguish between different redeemer types

Reply:
Sure, I can remove.